exercise-7/er.py

```diff
@@ -1,75 +1,86 @@
-
-def find_cs(path, cs,out, ):
-  with open(path ) as f:
-    texts = f.readlines ( )
+def find_cities(path, cs, output):
+  with open(path) as f:
+    texts = f.readlines()
     print(len(texts))
 
   # texts = texts[:50]
-  with open (out,'w') as g:
-    for t in texts:
-      c = t.split()
-      if c :
-        artid =c[ 0 ]
-        c = [w for w in c \
-            if w[0] not in [ "@", "<" ] ]
+  with open(output, 'w') as g:
+    for text in texts:
+      tokens = text.split()
 
+      if tokens:
+        article_id = tokens[0]
+        tokens = [token for token in tokens
+            if token[0] not in ["@", "<"]]
         counter = 0
-        for word in c:
-          if word in cs:
+        cities = []
```

Comment:
You get the dictionary of cities as the argument `cs` in `find_cities(path, cs, output)`, so you can check `if word in cs`. If you do `cities = []`, you are creating an empty list, and then later you check `if word in cities`, so if `word` is in the empty list you just created. Instead: Rename `find_cities(path, cities, output)`, delete `cities = []` and then check `if word in cities`.